app/components/specific/map/RaceMap.js

```diff
@@ -56,18 +57,38 @@ export default class RaceMap extends Component {
 
   static defaultProps = {};
 
+  constructor(props) {
+    super(props);
+    this.state = {
+      SAMPLE_REGION: {
+        latitude: this.props.race.path.coordinates[0].latitude || 47.478419,
+        longitude:
+          this.props.race.path.coordinates[0].longitude || -0.5631660000000238,
+        latitudeDelta: LATITUDE_DELTA,
+        longitudeDelta: LONGITUDE_DELTA
+      }
+    };
+  }
+
+  componentWillReceiveProps(nextProps) {
+    const { race } = this.props;
+    if (nextProps.race && nextProps.race.path !== race.path) {
+      this.setState({
```

Comment:
you can avoid to make your composant statefull

Reply:
I did not want to compute sample_region property each time render is called!

Reply:
i see your point but the wont happen i believe this is a too early optimisation